src/main/java/net/dries007/tfc/common/items/JavelinItem.java

```diff
@@ -6,15 +6,106 @@
 
 package net.dries007.tfc.common.items;
 
+import net.minecraft.core.BlockPos;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.sounds.SoundEvents;
+import net.minecraft.sounds.SoundSource;
+import net.minecraft.stats.Stats;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResultHolder;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.entity.projectile.AbstractArrow;
+import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.SwordItem;
 import net.minecraft.world.item.Tier;
+import net.minecraft.world.item.UseAnim;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.state.BlockState;
 
+import net.dries007.tfc.common.entities.ThrownJavelin;
+import net.dries007.tfc.util.Helpers;
+
+/**
+ * Modern implementation is based on {@link net.minecraft.world.item.TridentItem} rather than TFC Classic
+ */
 public class JavelinItem extends SwordItem
 {
-    public JavelinItem(Tier tier, float attackDamage, float attackSpeed, Properties properties)
+    private final ResourceLocation textureLocation;
+
+    public JavelinItem(Tier tier, float attackDamage, float attackSpeed, Properties properties, String name)
     {
         super(tier, (int) ToolItem.calculateVanillaAttackDamage(attackDamage, tier), attackSpeed, properties);
+        this.textureLocation = Helpers.identifier("textures/entity/projectiles/" + name + "_javelin.png");
```

Comment:
Should have two constructors: one which takes a string and passes `Helpers.identifier("textures/entity/projectile/" + name + "_javelin.png")` to the second constructor which takes a `ResourceLocation`